sklearn/tests/test_cross_validation.py

```diff
@@ -1155,6 +1155,55 @@ def bad_cv():
     assert_raises(ValueError, cval.cross_val_predict, est, X, y, cv=bad_cv())
 
 
+def test_cross_val_predict_apply():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    classes = len(set(y))
+
+    # Shuffle the data before doing cross validation
+    shf = np.random.permutation(np.arange(len(X)))
+    X = X[shf]
+    y = y[shf]
+
+    cv = cval.KFold(len(iris.target))
```

Comment:
what happens if you don't use kfolds and a samples appears multiple times in test ?